### ISSUE TITLE
Can't install SnowCross 2000 game

### ISSUE THREAD

**Issue by JamesJacquesDiego** [danoon2/Boxedwine#3]
Hi,
I tried to install SnowCross 2000 in Boxedwine ( this game https://www.youtube.com/watch?v=FLlSYboTCXI ). Unfortunately I couldn't install this game. Last time I could was on win 7 with xp mode.  I'd be glad if you could check it on win 10.

Sorry for posting it here but I couldn't find your email to ask for that.

**Comment by danoon2** [comment]
So if I understand you correct, the game starts just fine and when you go through the menus and select the bike and track, after you hit the next button the window disappears and that is it?

It sounds like the game is about to transition to the main game windows so it might have changed resolutions.  I'm assuming you are on Windows running the x64 version of Boxedwine 20.1.2?  Are you running it from the Boxedwine UI or from the command line?  Are you familiar with Task Manager on Windows?  After it crashes can you check to see if boxedwine.exe is still running and if it is using CPU time?

**Comment by JamesJacquesDiego** [comment]
Yes and no. I have two versions of this game. First is from my cd which I bought 20 years ago and when I installed this game on win 7 I copied folder with the game. This version sometimes works when i try to run it. Second version is the installer of this game from internet which i downloaded when I tried to run on windows 10 in the past. this version stops working when trying to install.

I'm running it from the Boxedwine UI (x64). After it crashes, it disappears from the application list in task manager, but another process appears in the list of background processes (two processes in the background - before crash was one). After a few tries and crashes I've had 9 processes in the background and 0 in applications which are still running.

If you give me your email I can send you two links to .zip files with game folders on my google drive to check.

**Comment by JamesJacquesDiego** [comment]
Second version I installed on win 10 and then upload folder to boxedwine. Installation was succesful but when game started it crashed with message: "A program on your system has crashed, but WineDbg was unable to attach to the process to obtain a backtrace"

**Comment by danoon2** [comment]
When boxedwine.exe is started with no arguments, it starts the UI.  When a game is launched from the UI it starts another boxedwine.exe for the game and hides the UI windows (so 2 processes are running).  It seems like the game is switching from 640x480 in the menus to 1024x768 for the game.  After it dismisses the 640x480 window it sometimes doesn't create the 1024x768 window.   This is why there are 2 background processes in in the task manager, more if you tried more times.  The process running the game simply got stuck in some sort of loop without a window.  You should kill those if you see them.

How often does this fail for you?  Have you seen it work at all?  For me it fails about 20% of the time.  What CPU are you using?  I'm wondering if this is a timing issue and I should test on a slower or faster computer.

**Comment by JamesJacquesDiego** [comment]
I use asus rog gl503vm with intel i7-7700HQ processor. This game fails each time. I couldn't run the game. I tried use wine 5.0 then 1.7 and nothing changed. Sometimes it stops on installing wine installer or gecko installer also (when I tried to install it again). But in most situations is like you say and it probably crashes when switching resolution.  It is happening each time when I try run this game.

**Comment by danoon2** [comment]
Seems like the game is going into an infinite loop in this function in wined3d

static void wined3d_cs_mt_finish(struct wined3d_cs *cs, enum wined3d_cs_queue_id queue_id)
{
    if (cs->thread_id == GetCurrentThreadId())
        return wined3d_cs_st_finish(cs, queue_id);

    while (cs->queue[queue_id].head != *(volatile LONG *)&cs->queue[queue_id].tail)
        wined3d_pause();
}

I've tried setting CPU affinity in Boxedwine so that Boxedwine would be confined to a single CPU (this helps some games around 1999-2002 or so because they had multi threaded support but machines were single CPU) and I tried setting csmt to disabled as specified here: https://wiki.winehq.org/Useful_Registry_Keys  It seemed like this improved things, but I might have just been unlucky while reproducing it since it doesn't happen every time for me.  I also tried Wine 4.0 and Wine 5.0/Win95

For now I will keep this issue open, but I'm out of ideas for where the bug is.  It could be in the game, in Wine or in Boxedwine.

**Comment by JamesJacquesDiego** [comment]
Thank you for your help. I don't think the problem is with wine because I ran this game with wine on virtualbox(xubuntu with wine (probably 1.7 I don't remember)) and on wine on macbook air. On virtualbox there were problems with sound and lagging. On MacBook it was quite good but cooling system was not enough and macbook was really hot.
